pkg/config/security/security.go

```diff
@@ -217,6 +229,34 @@ func IsValidCipherSuite(cs string) bool {
 	return ValidCipherSuites.Contains(cs)
 }
 
+func IsValidEcdhCurve(cs string) bool {
```

Comment:
This is a little confusing to me - if no curves are specified, that's a valid curve?

What's the expected/desired outcome in this case? All curves allowed? No curves allowed?

If this does have special meaning, an explicit test for no curves specified would be good.

Reply:
MeshConfig ECDH curves are validated only if atleast one curve is specified. If no curves are specified it follows the current default behavior for backward compatibility - which means we let envoy decide the default values for us, as specified in : https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto

Reply:
Makes sense - a comment indicating that for this function would be good, as it's not entirely obvious that is what would happen from this code.

Reply:
@bleggett I added this to the description of the field in the API PR, so that people who knows the config knows how to use it and what the defaults are

Reply:
That works, thx! As long as it's breadcrumb'd.